### PR TITLE
Remove Timeout Parameter

### PR DESCRIPTION
**Pull request by adithyant**

### CODE REVIEW

File: mmd/mmd.go [hunk 1]
```diff
@@ -100,7 +100,8 @@ func (c *Conn) reconnect() {
 
 	err = c.createSocketConnection()
 	if err != nil {
-		log.Panicln("Failed to reconnect: ", err)
+		log.Println("Failed to reconnect: ", err)
```
Comment (dbudworth) [inline mmd/mmd.go]: can this even happen?    now that you spin on error, a retryable connection can't return error

File: mmd/mmd.go [hunk 2]
```diff
@@ -111,11 +112,10 @@ func (c *Conn) createSocketConnection() error {
 		return err
 	}
 
-	start := time.Now()
 	for {
 		conn, err := net.DialTCP("tcp", nil, addr)
-		if err != nil && c.config.AutoRetry && time.Since(start) < c.config.Timeout {
-			time.Sleep(5 * time.Second)
+		if err != nil && c.config.AutoRetry {
+			time.Sleep(c.config.ReconnectInterval)
```
Comment (dbudworth) [inline mmd/mmd.go]: "Disconnected" isn't right thing to say.
probable want something like:
log.Println("Failed to connect:",err)

And, if you've had at least 1 failure, you should log success.
basically, if your loop had attempt++
then, on success:
if attempt > 1 { log.Println("reconnected") }
so someone who reads the log sees it failed for a while then it succeeded

File: mmd/mmd.go (hunk 1 above)
Comment (adithyant) [reply mmd/mmd.go]: That's true. I will remove this.

File: mmd/mmd.go (hunk 2 above)
Comment (adithyant) [reply mmd/mmd.go]: I'm logging the "reconnected" message after the connection is made at line 106

File: mmd/mmd.go [hunk 3]
```diff
@@ -98,11 +98,11 @@ func (c *Conn) reconnect() {
...
 
+	start := time.Now()
 	err = c.createSocketConnection()
-	if err != nil {
-		log.Panicln("Failed to reconnect: ", err)
-	}
-	log.Println("Socket reset. Reconnected to mmd")
+	elapsed := time.Since(start)
+
+	log.Println("Socket reset. Connected to mmd after : ", elapsed)
```
Comment (dbudworth) [inline mmd/mmd.go]: in go, all `Println` functions automatically pad with a space between arguments.
This log line would print:
`after<space>:<space><space>10s`

change to:
`log.Println("Socket reset. Connected to mmd after :", elapsed)`

File: mmd/mmd.go [hunk 4]
```diff
@@ -111,12 +111,11 @@ func (c *Conn) createSocketConnection() error {
...
 
-	start := time.Now()
 	for {
 		conn, err := net.DialTCP("tcp", nil, addr)
-		if err != nil && c.config.AutoRetry && time.Since(start) < c.config.Timeout {
-			time.Sleep(5 * time.Second)
-			log.Println("Disconnected. Retrying again")
+		if err != nil && c.config.AutoRetry {
+			time.Sleep(c.config.ReconnectInterval)
+			log.Println("Failed to connect : ", err)
```
Comment (dbudworth) [inline mmd/mmd.go]: same comment on spacing in a `Println`

**Pull request merged**